tensorflow/python/autograph/impl/api.py

```diff
@@ -721,6 +721,10 @@ def to_code_v1(entity,
   del arg_values
   del arg_types
   del indentation
+  if not hasattr(type(entity), '__code__'):
```

Comment:
I fear performing this check here is too early - these is logic to unpack partials, method-wrappers, and others. A better place to raise the error is just before accessing the __code__ field.

Reply:
Okay. So just delete this one and leave the other one on the bottom?

Reply:
Rather, adding it in conversion.py at line 310, just before it tries to make the offending call, would be better. At that point, we can't tell whether the caller is to_code, to_graph or something else. So the message should be more generic a bit:

"[...]. If this is a @tf.function, try passing f.python_function instead."

Reply:
Ah I see. So the corrent place to put it is [here](https://github.com/tensorflow/tensorflow/blob/63fdf5ffe374e408d85ac0075fc51b8677a29f52/tensorflow/python/util/tf_inspect.py#L345)?  

Like this?
```
  if not hasattr(type(tf_decorator.unwrap(object)[1]), '__code__'):
    raise ValueError('Cannot apply autograph to a function that doesn\'t '
                     'expose a __code__ object. If this is a @tf.function,'
                     ' please use to_code(f.python_function) instead.')
```

Reply:
Alright, thank you!